cli/scripts/e2e/e2e.ts

```diff
@@ -657,13 +655,3 @@ function getCachedLogFilePathForTask(
 ): string {
   return path.join(cacheDir, pathToPackage, ".turbo", `turbo-${taskName}.log`);
 }
-
-// Setup corepack and provide the dir to the binaries that corepack installs
-// It is important that this dir has a higher priority in $PATH than any other
-// dirs that may versions of package managers installed by other mechanisms
-// e.g. homebrew or pnpm
-function setupCorepack(): string {
-  execa.sync("corepack", ["enable"]);
-  const corepack_path = execa.sync("which", ["corepack"]).stdout;
-  return path.dirname(corepack_path);
-}
```

Comment:
If we're getting rid of this, then we should add a section to `CONTRIBUTING.md` that alerts users that they need to make sure their $PATH is correctly setup.
For example, with my setup this change causes `make e2e` to hard fail (I do have corepack enabled):
```
olszewski@chriss-mbp cli % echo $PATH
/Users/olszewski/Library/pnpm:/opt/homebrew/bin:/opt/homebrew/sbin:/Users/olszewski/.nvm/versions/node/v16.16.0/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/homebrew/bin:/opt/homebrew/sbin:/Users/olszewski/.nvm/versions/node/v16.16.0/bin:/usr/local/go/bin:/Users/olszewski/go/bin:/usr/local/go/bin:/Users/olszewski/go/bin
olszewski@chriss-mbp cli % make e2e
npm install -g corepack@latest
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.

changed 1 package, and audited 2 packages in 440ms

found 0 vulnerabilities
corepack enable
pnpm install --filter=cli
..                                       |   +5 +
..                                       | Progress: resolved 5, reused 5, downloaded 0, added 0, done
node -r esbuild-register scripts/e2e/e2e.ts
Error: Command failed with exit code 1: yarn install
error Couldn't find any versions for "b" that matches "workspace:*"
yarn install v1.22.19
[1/4] Resolving packages...
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
    at makeError (/Users/olszewski/code/vercel/turborepo/node_modules/.pnpm/execa@5.1.1/node_modules/execa/lib/error.js:60:11)
    at Function.module.exports.sync (/Users/olszewski/code/vercel/turborepo/node_modules/.pnpm/execa@5.1.1/node_modules/execa/index.js:194:17)
    at Monorepo.linkPackages (/Users/olszewski/code/vercel/turborepo/cli/scripts/monorepo.ts:162:17)
    at Object.<anonymous> (/Users/olszewski/code/vercel/turborepo/cli/scripts/e2e/e2e.ts:52:8)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Module._compile (/Users/olszewski/code/vercel/turborepo/node_modules/.pnpm/esbuild-register@3.3.3_esbuild@0.14.49/node_modules/esbuild-register/dist/node.js:2258:26)
    at Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Object.newLoader [as .ts] (/Users/olszewski/code/vercel/turborepo/node_modules/.pnpm/esbuild-register@3.3.3_esbuild@0.14.49/node_modules/esbuild-register/dist/node.js:2262:9)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12) {
  shortMessage: 'Command failed with exit code 1: yarn install',
  command: 'yarn install',
  escapedCommand: 'yarn install',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: 'yarn install v1.22.19\n' +
    '[1/4] Resolving packages...\n' +
    'info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.',
  stderr: `error Couldn't find any versions for "b" that matches "workspace:*"`,
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
make: *** [e2e] Error 1
olszewski@chriss-mbp cli % which yarn
/opt/homebrew/bin/yarn
olszewski@chriss-mbp cli % which pnpm
/opt/homebrew/bin/pnpm
olszewski@chriss-mbp cli % pnpm -v
7.8.0
```

Reply:
Summary of a conversation that happened in Slack:

We decided that we should stay ahead of where things are going with `corepack` and begin adoption of it. There exist weird `$PATH` precedence issues that are not fun and it's still behind `corepack enable`, but we believe that those issues will fade as adoption of `corepack` increases and it moves toward being the default for `node` dev environments.

It's imperfect, it will make things a little weird for contributors during the transition, but we believe that it's worth the churn and add or name to the "adopters of `corepack`" list.